tests/Security/TwoFactor/Trusted/TrustedCookieResponseListenerTest.php

```diff
@@ -200,6 +191,59 @@ public function provideRequestHostName(): array
             ['2001:0db8:85a3:0000:0000:8a2e:0370:7334'],
         ];
     }
+
+    /**
+     * @test
+     */
+    public function onKernelResponse_cookieSecureAuto_secureRequest(): void
+    {
+        $this->trustedTokenStorage
+            ->expects($this->any())
+            ->method('hasUpdatedCookie')
+            ->willReturn(true);
+
+        $this->trustedTokenStorage
+            ->expects($this->any())
+            ->method('getCookieValue')
+            ->willReturn('cookieValue');
+
+        $request = $this->createRequestWithHost();
+        $request->server->set('HTTPS', 'On');
+
+        $event = $this->createEventWithRequest($request);
+        $this->cookieResponseListener = $this->createTrustedCookieResponseListener(null, null);
+        $this->cookieResponseListener->onKernelResponse($event);
+
+        $cookies = $this->response->headers->getCookies();
+        $this->assertCount(1, $cookies, 'Response must have a cookie set.');
+        $this->assertTrue($cookies[0]->isSecure());
+    }
+
+    /**
+     * @test
+     */
+    public function onKernelResponse_cookieSecureAuto_unsecureRequest(): void
```

Comment:
Can we rename please? `onKernelResponse_cookieSecureAutoOnUnsecureRequest_setUnsecureCookie`